File: mianyang-campus/backend/app/schemas/growth.py
from pydantic import BaseModel
from datetime import date


class GrowthRecordCreate(BaseModel):
    type: str
    title: str
    description: str | None = None
    date: str
    attachment_url: str | None = None
    # 荣誉
    honor_level: str | None = None
    # 竞赛
    organizer: str | None = None
    competition_level: str | None = None
    # 实践
    practice_type: str | None = None
    practice_certificate: str | None = None
    # 论文
    paper_type: str | None = None
    paper_name: str | None = None
    first_author: str | None = None
    second_author: str | None = None
    third_author: str | None = None
    # 成果
    achievement_type: str | None = None
    achievement_name: str | None = None


class GrowthRecordOut(BaseModel):
    id: int
    student_id: int
    type: str
    title: str
    description: str | None = None
    date: date
    attachment_url: str | None = None
    # 荣誉
    honor_level: str | None = None
    # 竞赛
    organizer: str | None = None
    competition_level: str | None = None
    # 实践
    practice_type: str | None = None
    practice_certificate: str | None = None
    # 论文
    paper_type: str | None = None
    paper_name: str | None = None
    first_author: str | None = None
    second_author: str | None = None
    third_author: str | None = None
    # 成果
    achievement_type: str | None = None
    achievement_name: str | None = None

    class Config:
        from_attributes = True


class RadarDimension(BaseModel):
    name: str
    value: float


class MonthlyStat(BaseModel):
    month: str
    count: int
    type: str


class GpaPoint(BaseModel):
    semester: str
    gpa: float


class GrowthProfileOut(BaseModel):
    total_score: float
    radar: list[RadarDimension]
    stats_by_type: list[dict]
    monthly_trend: list[MonthlyStat]
    skills: list[str]
    interests: list[str]
    total_records: int
    total_skills: int
    gpa_trend: list[GpaPoint]


class StudentProjectCreate(BaseModel):
    project_name: str
    start_date: str
    end_date: str | None = None
    is_team: bool = False
    team_members: str | None = None
    attachment_url: str | None = None


class StudentProjectOut(BaseModel):
    id: int
    student_id: int
    project_name: str
    start_date: str
    end_date: str | None = None
    is_team: bool
    team_members: str | None = None
    attachment_url: str | None = None

    class Config:
        from_attributes = True

File: mianyang-campus/frontend/src/api/growth.ts
import request from '@/utils/request'
import type { GrowthRecord } from '@/types'

export interface GrowthProfile {
  total_score: number
  radar: { name: string; value: number }[]
  stats_by_type: { name: string; value: number }[]
  monthly_trend: { month: string; count: number; type: string }[]
  skills: string[]
  interests: string[]
  total_records: number
  total_skills: number
  gpa_trend: { semester: string; gpa: number }[]
}

export function getGrowthProfile() {
  return request.get<GrowthProfile>('/growth/profile')
}

export function getGrowthRecords(studentId?: number) {
  return request.get<GrowthRecord[]>('/growth/records', { params: { student_id: studentId } })
}

export function createGrowthRecord(data: Partial<GrowthRecord>) {
  return request.post<GrowthRecord>('/growth/records', data)
}

export function deleteGrowthRecord(id: number) {
  return request.delete(`/growth/records/${id}`)
}

// ---- Project Showcase ----
export interface StudentProject {
  id: number
  student_id: number
  project_name: string
  start_date: string
  end_date: string | null
  is_team: boolean
  team_members: string | null
  attachment_url: string | null
}

export function getProjects() {
  return request.get<StudentProject[]>('/growth/projects')
}

export function createProject(data: Partial<StudentProject>) {
  return request.post<StudentProject>('/growth/projects', data)
}

export function updateProject(id: number, data: Partial<StudentProject>) {
  return request.put<StudentProject>(`/growth/projects/${id}`, data)
}

export function deleteProject(id: number) {
  return request.delete(`/growth/projects/${id}`)
}

export function updateSkills(data: { skills: string[]; interests: string[] }) {
  return request.put('/growth/skills', data)
}
